### PR TITLE
Rhys/key rotation

### PR DESCRIPTION
**Pull request by wheresrhys**

### CODE REVIEW

File: bin/fastly-tools.js [hunk 1]
```diff
@@ -15,13 +15,14 @@ program
...
 	.option('-b --backends <backends>', 'Upload the backends specified in <backends> via the api')
+	.option('-k --api-keys <keys>', 'list of alternate api keys to try should the key stored in process.env.FASTLY_API_KEY hit its rate limit', list)
 	.action(function(folder, options) {
 		const deploy = require('../tasks/deploy');
 		const log = require('../lib/logger')({verbose:options.verbose, disabled:options.disableLogs});
 		const exit = require('../lib/exit')(log, true);
 
 		const symbols = require('../lib/symbols');
-
+		console.log(options.apiKeys);
```
Comment (JakeChampion) [inline bin/fastly-tools.js]: this would leak api keys to log files

File: README.md [hunk 2]
```diff
@@ -27,14 +27,15 @@ npm install --save-dev @financial-times/fastly-tools
...
-	  Options:
-	
-		-h, --help               output usage information
-		-m, --main <main>        Set the name of the main vcl file (the entry point).  Defaults to "main.vcl"
-		-v, --vars <vars>        A way of injecting environment vars into the VCL.  So if you pass --vars AUTH_KEY,FOO the values {$AUTH_KEY} and ${FOO} in the vcl will be replaced with the values of the environmemnt variable.  If you include SERVICEID it will be populated with the current --service option
-		-e, --env                Load environment variables from local .env file (use when deploying from a local machine
-		-s, --service <service>  REQUIRED.  The ID of the fastly service to deploy to.
-		-V --verbose             Verbose log output
-		b --backends <backends>  Upload the backends specified in <backends> via the api
+  Usage: deploy [options] [folder]
```
Comment (JakeChampion) [inline README.md]: this was indented as it is part of the help output

File: lib/fastly/lib/index.js [hunk 3]
```diff
@@ -9,16 +9,17 @@
...
 /**
  * Constructor
  */
-function Fastly (apikey, service, options) {
+function Fastly (apikeys, service, options) {
 
 	var opts = options || {};
-	this.apikey = apikey || '';
+	this.apikey = apikeys.shift();
+	this.apikeys = apikeys || [];
```
Comment (JakeChampion) [inline lib/fastly/lib/index.js]: apikeys should be defaulted to an array before any array commands are called on it. 
```
this.apikeys = apikeys || [];
this.apikey = apikeys.shift();
```

File: lib/fastly/lib/index.js [hunk 4]
```diff
@@ -9,16 +9,17 @@
...
  */
 const request = require('request');
-const Q = require('q');
 const debug = require('debug')('fastly');
 
 /**
  * Constructor
  */
-function Fastly (apikey, service, options) {
+function Fastly (apikeys, service, options) {
```
Comment (JakeChampion) [inline lib/fastly/lib/index.js]: If passed a single apikey, we should wrap in up in an array

File: README.md (hunk 2 above)
Comment (wheresrhys) [inline README.md]: I just copy pasted from the help output

File: lib/fastly/lib/index.js [hunk 5]
```diff
@@ -70,47 +70,53 @@ function Fastly (apikey, service, options) {
...
 
-		deferred.resolve(body);
-	});
+			resolve(body);
+		});
+
+	})
+		.catch(err => {
+			if (err.message === 'api rate limit' && this.apiKeys.length) {
+				this.apikey = this.apikeys.shift();
```
Comment (JakeChampion) [inline lib/fastly/lib/index.js]: 👍

File: bin/fastly-tools.js (hunk 1 above)
Comment (wheresrhys) [inline bin/fastly-tools.js]: ah yes, meant to remove it

File: lib/fastly/lib/index.js (hunk 4 above)
Comment (wheresrhys) [inline lib/fastly/lib/index.js]: this is an internal class, so I think just accepting an array is ok

File: tasks/deploy.js [hunk 6]
```diff
@@ -32,18 +33,22 @@ function task (folder, opts) {
...
 
-		if (!process.env.FASTLY_APIKEY) {
-			throw new Error('FASTLY_APIKEY not found');
+		if (process.env.FASTLY_APIKEY) {
+			options.apiKeys.unshift(process.env.FASTLY_APIKEY);
 		}
 
-		const fastlyApiKey = process.env.FASTLY_APIKEY;
+		if (!options.apiKeys.length) {
+			throw new Error('fastly api key not found. Wither set a FASTLY_APIKEY environment variable, or pass in using the --api-keys option');
```
Comment (JakeChampion) [inline tasks/deploy.js]: Wither --> Either

File: lib/fastly/lib/index.js (hunk 3 above)
Comment (wheresrhys) [inline lib/fastly/lib/index.js]: again, it's an internal class, and the task that calls it takes care of ensuring it's an array... but I supose there's no harm in adding this

**Comment by JakeChampion** [comment]
Seem to have 1 failing test on ci, other than that, looks good

**Pull request merged**